src/features/query-bar/__tests__/FilterButton-test.js

```diff
@@ -21,163 +22,115 @@ describe('feature/query-bar/components/filter/FilterButton', () => {
         });
     });
 
-    describe('handleFieldChange()', () => {
+    describe('handleColumnChange()', () => {
         [
             {
                 description: 'should set conditions with an object with column',
-                index: 0,
+                columnId: '2',
                 condition: {
+                    columnId: '1',
                     id: '3',
-                    columnDisplayText: '',
-                    columnKey: null,
-                    operatorDisplayText: '',
-                    operatorKey: 0,
-                    valueDisplayText: null,
-                    valueKey: null,
+                    operator: IS,
+                    value: null,
                 },
                 conditions: [
                     {
+                        columnId: '1',
                         id: '3',
-                        columnDisplayText: '',
-                        columnKey: null,
-                        operatorDisplayText: '',
-                        operatorKey: 0,
-                        valueDisplayText: null,
-                        valueKey: null,
+                        operator: IS,
+                        value: null,
                     },
                 ],
-                fieldDisplayText: 'myColumn',
-                fieldDisplayTextType: 'columnDisplayText',
-                fieldKey: 0,
-                fieldKeyType: 'columnKey',
-                valueType: 'string',
-                updatedCondition: [
+                newCondition: [
                     {
+                        columnId: '2',
                         id: '3',
-                        columnDisplayText: 'myColumn',
-                        columnKey: 0,
-                        operatorDisplayText: '',
-                        operatorKey: 0,
-                        valueDisplayText: null,
-                        valueKey: null,
-                        valueType: 'string',
+                        operator: IS,
+                        value: null,
                     },
                 ],
             },
+        ].forEach(({ description, columnId, condition, conditions, newCondition }) => {
+            test(`${description}`, () => {
+                const wrapper = getWrapper({ columns });
+                wrapper.setState({
+                    conditions,
```

Comment:
This name is confusing and I cna't distinguish from condition / newCondition but understandable if you want to fix in a separate PR.

Reply:
Resolving in a separate PR

Reply:
Would renaming it like this be better?

conditions / expectedConditions

Reply:
Oops sorry I missed your question. I've filtered my github emails because it's so spammy.